common/pkg/cgroups/cgroups_linux.go

```diff
@@ -92,63 +84,37 @@ func init() {
 }
 
 // getAvailableControllers get the available controllers.
-func getAvailableControllers(exclude map[string]controllerHandler, cgroup2 bool) ([]controller, error) {
-	if cgroup2 {
-		controllers := []controller{}
-		controllersFile := filepath.Join(cgroupRoot, "cgroup.controllers")
-
-		// rootless cgroupv2: check available controllers for current user, systemd or servicescope will inherit
-		if unshare.IsRootless() {
-			userSlice, err := getCgroupPathForCurrentProcess()
-			if err != nil {
-				return controllers, err
-			}
-			// userSlice already contains '/' so not adding here
-			basePath := cgroupRoot + userSlice
-			controllersFile = filepath.Join(basePath, "cgroup.controllers")
-		}
-		controllersFileBytes, err := os.ReadFile(controllersFile)
-		if err != nil {
-			return nil, fmt.Errorf("failed while reading controllers for cgroup v2: %w", err)
-		}
-		for controllerName := range strings.FieldsSeq(string(controllersFileBytes)) {
-			c := controller{
-				name:    controllerName,
-				symlink: false,
-			}
-			controllers = append(controllers, c)
-		}
-		return controllers, nil
-	}
-
-	subsystems, _ := cgroupV1GetAllSubsystems()
+func getAvailableControllers() ([]controller, error) {
 	controllers := []controller{}
-	// cgroupv1 and rootless: No subsystem is available: delegation is unsafe.
-	if unshare.IsRootless() {
-		return controllers, nil
-	}
+	controllersFile := filepath.Join(cgroupRoot, "cgroup.controllers")
 
-	for _, name := range subsystems {
-		if _, found := exclude[name]; found {
-			continue
-		}
-		fileInfo, err := os.Stat(cgroupRoot + "/" + name)
+	// rootless cgroupv2: check available controllers for current user, systemd or servicescope will inherit
+	if unshare.IsRootless() {
+		userSlice, err := getCgroupPathForCurrentProcess()
 		if err != nil {
-			continue
+			return controllers, err
 		}
+		// userSlice already contains '/' so not adding here
+		basePath := cgroupRoot + userSlice
+		controllersFile = filepath.Join(basePath, "cgroup.controllers")
+	}
+	controllersFileBytes, err := os.ReadFile(controllersFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed while reading controllers for cgroup v2: %w", err)
+	}
+	for controllerName := range strings.FieldsSeq(string(controllersFileBytes)) {
 		c := controller{
-			name:    name,
-			symlink: !fileInfo.IsDir(),
+			name:    controllerName,
+			symlink: false,
```

Comment:
`symlink` is always `false` (and never read?)

… and then I think the only remaining caller only extracts the `name` field, so maybe the whole `controller` type can be removed.